typescript-typing-tutor/main.ts

```diff
@@ -0,0 +1,38 @@
+document.addEventListener('DOMContentLoaded', () => {
+  const spans = document.querySelectorAll('#phrase span');
+  let currentIndex = 0;
+
+  function updateCurrentSpan(): void {
+    spans.forEach((span, index): void => {
+      if (index === currentIndex) {
+        span.classList.add('current');
+      } else {
+        span.classList.remove('current');
+      }
+    });
+  }
+
+  function handleKeydown(event: KeyboardEvent): void {
+    if (currentIndex >= spans.length) return;
+    const currentSpan = spans[currentIndex];
+    const expectedChar = currentSpan.textContent;
+    const typedChar = event.key;
+
+    if (typedChar === expectedChar) {
+      currentSpan.classList.add('correct');
+      currentIndex++;
+    } else {
+      currentSpan.classList.add('incorrect');
+    }
+
+    setTimeout(() => {
+      currentSpan.classList.remove('incorrect');
+      updateCurrentSpan();
+    }, 500);
+
```

Comment:
We only want the `incorrect` class to be removed when the user gets the answer right, not based on time.